src/Core/Language/Pack/Loader/RemoteLanguagePackLoader.php

```diff
@@ -51,19 +55,41 @@ public function __construct(Version $version)
         $this->version = $version;
     }
 
+    /**
+     * @param string $locale if no locale given, return URL for available languages list
+     *
+     * @return string requested URL
+     */
+    public function getLanguagePackUrl(?string $locale = null): string
+    {
+        if ($locale) {
+            return self::LANG_REPOSITORY_BASE_PATH . '/translations/' . $this->version->getSemVersion() . '/' . $locale . '/' . $locale . '.zip';
+        } else {
+            return self::LANG_REPOSITORY_BASE_PATH . '/translations/' . $this->version->getSemVersion() . '/available_languages.json';
+        }
+    }
+
     /**
      * {@inheritdoc}
      */
     public function getLanguagePackList()
     {
-        $normalizedLink = str_replace('%ps_version%', $this->version->getSemVersion(), self::PACK_LINK);
+        $normalizedLink = $this->getLanguagePackUrl();
         $jsonResponse = file_get_contents($normalizedLink);
-
-        $result = [];
-        if ($jsonResponse) {
-            $result = json_decode($jsonResponse, true);
-        }
+        $result = json_decode($jsonResponse, true) ?? [];
 
         return $result;
     }
+
+    /**
+     * Builds an instance from version defined in environment (`_PS_VERSION_`) _
+     *
+     * @return self
+     */
+    public static function buildFromEnv()
```

Comment:
Not sure that the name of this function is the best one. I don't think `_PS_VERSION_` environment-dependent 🤔
Maybe something like `buildFromConst()` or even just `build()` makes more sense to me